### PR TITLE
CA-316: add per hour calculation logic to LabourCostFormBloc

### PR DESCRIPTION
**Pull request by ayanasamuel8**
### 1. PR SUMMARY

Adds per-hour calculation logic to `LabourCostFormBloc`. Introduces four new events (`LabourCostCalculationMethodChanged`, `LabourCostLaborValueUpdated`, `LabourCostCrewSizeUpdated`, `LabourCostHourlyRateUpdated`) with handlers that store their values in `LabourCostFormEditing` state and expose a `perHourTotal` getter (`hourlyRate × laborHours × crewSize`). No UI changes; the widget wiring will follow with the submission ticket (CA-294). 21 unit tests cover each event's validation and the total calculation across method types.

### 2. REVIEW SUMMARY TABLE

No issues found.

## Test plan
- [ ] `fvm flutter test test/features/estimations/units/blocs/labour_cost_form_bloc_test.dart` — all 21 tests pass
- [ ] `fvm dart run custom_lint` — no issues
- [ ] `fvm flutter test` — no new failures (4 pre-existing failures in tag/app-shell tests unchanged)

### CODE REVIEW

File: lib/features/estimation/presentation/bloc/labour_cost_form_bloc/labour_cost_form_bloc.dart [hunk 1]
```diff
@@ -24,4 +29,66 @@ class LabourCostFormBloc
...
+    Emitter<LabourCostFormState> emit,
+  ) {
+    final current = state is LabourCostFormEditing
+        ? state as LabourCostFormEditing
+        : const LabourCostFormEditing();
+    final parsed = double.tryParse(event.value);
+    final isPositive = parsed != null && parsed > 0;
+    emit(
+      current.copyWith(
+        hourlyRate: isPositive ? parsed : null,
```
Comment (ripplearcgit) [inline lib/features/estimation/presentation/bloc/labour_cost_form_bloc/labour_cost_form_bloc.dart:86]: 🟡 **Reset-to-null transition on invalid input isn't test-covered**

Both `_onCrewSizeUpdated` and `_onHourlyRateUpdated` reset the stored value to `null` when the new input is invalid/non-positive (rather than preserving the previous valid value). This is defensible — it keeps `perHourTotal` from silently computing against a stale value once the field is in an error state — but none of the 21 tests exercise the "previously valid → now invalid" transition: every existing test dispatches a single event from the bloc's fresh/initial state, where `crewSize`/`hourlyRate` is already `null`. A mutation that changed the null branch to `isPositive ? crewSize : current.crewSize` (preserve stale value instead of clearing it) would not be caught by any test today. The same gap applies to the mirrored test file (`labour_cost_form_bloc_test.dart`).

```dart
blocTest<LabourCostFormBloc, LabourCostFormState>(
  'resets hourlyRate to null when a previously valid value becomes invalid',
  build: () => bloc,
  act: (bloc) => bloc
    ..add(const LabourCostHourlyRateUpdated('100'))
    ..add(const LabourCostHourlyRateUpdated('-5')),
  expect: () => [
    isA<LabourCostFormEditing>().having((s) => s.hourlyRate, 'hourlyRate', 100.0),
    isA<LabourCostFormEditing>().having((s) => s.hourlyRate, 'hourlyRate', isNull),
  ],
);
```

File: lib/features/estimation/presentation/bloc/labour_cost_form_bloc/labour_cost_form_event.dart [hunk 2]
```diff
@@ -10,4 +10,28 @@ class LabourCostItemTypeChanged extends LabourCostFormEvent {
...
+class LabourCostCrewSizeUpdated extends LabourCostFormEvent {
+  const LabourCostCrewSizeUpdated(this.crewSize);
+
+  final double? crewSize;
+}
+
+class LabourCostHourlyRateUpdated extends LabourCostFormEvent {
+  const LabourCostHourlyRateUpdated(this.value);
+
+  final String value;
```
Comment (ripplearcgit) [inline lib/features/estimation/presentation/bloc/labour_cost_form_bloc/labour_cost_form_event.dart:37]: 🟡 **crewSize vs hourlyRate event-contract asymmetry**

`LabourCostCrewSizeUpdated` takes a pre-parsed `double?` while `LabourCostHourlyRateUpdated` takes a raw `String` parsed bloc-side. Both back analogous numeric text-field inputs, so the asymmetry means CA-294's widget wiring will need two different dispatch patterns for what are otherwise parallel fields — e.g. the crew-size field's `TextEditingController` text must be pre-parsed by the widget before dispatch, while the hourly-rate field's raw text is passed straight through. Suggest picking one convention (raw `String` parsed bloc-side is more consistent with the existing `itemType`/`hourlyRate` precedent) before CA-294 starts.

File: lib/features/estimation/presentation/bloc/labour_cost_form_bloc/labour_cost_form_state.dart [hunk 3]
```diff
@@ -9,22 +9,64 @@ class LabourCostFormInitial extends LabourCostFormState {
...
+    this.crewSize,
+    this.crewSizeError,
+    this.hourlyRate,
+    this.hourlyRateError,
+  });
 
   final String labourType;
   final String? itemTypeError;
+  final LaborCalculationMethodType? calculationMethod;
+  final LaborValue? laborValue;
```
Comment (ripplearcgit) [inline lib/features/estimation/presentation/bloc/labour_cost_form_bloc/labour_cost_form_state.dart:32]: 🟡 **British/American spelling mix in `laborValue`**

`laborValue` (the field) mixes American "Labor" (from the pre-existing `LaborValue`/`LaborCalculationMethodType` entity naming) with the surrounding British "Labour" convention used everywhere else in this file (`LabourCostFormEditing`, `labourType`). Inherited from the pre-existing entity type name, not newly introduced spelling, but now baked into a new public event class name (`LabourCostLaborValueUpdated`) too — worth a one-time cleanup pass if the team standardizes on one spelling.

**Review by ripplearcgit — APPROVED** [review]
### Architectural overview

```mermaid
sequenceDiagram
    participant W as Widget (CA-294, not yet wired)
    participant B as LabourCostFormBloc
    participant S as LabourCostFormEditing

    Note over W,B: Deferred: widget still uses local TextEditingControllers (CA-353)
    W-->>B: future: add(LabourCostHourlyRateUpdated('100'))
    B->>B: parsed = double.tryParse('100') = 100.0, isPositive = true
    B->>S: copyWith(hourlyRate: 100.0, hourlyRateError: null)
    W-->>B: future: add(LabourCostCrewSizeUpdated(2.0))
    B->>S: copyWith(crewSize: 2.0, crewSizeError: null)
    W-->>B: future: add(LabourCostLaborValueUpdated(LaborValue laborHours 8.0))
    B->>S: copyWith(laborValue: laborHours 8.0)
    W-->>B: future: add(LabourCostCalculationMethodChanged(perHour))
    B->>S: copyWith(calculationMethod: perHour)
    Note over S: perHourTotal getter reads current state
    alt calculationMethod != perHour
        S->>S: return 0
    else calculationMethod == perHour
        S->>S: return hourlyRate * laborHours * crewSize = 100 * 8 * 2 = 1600
    end
```

**Verdict: Conditional LGTM** — bloc-only scaffolding for per-hour labour cost calculation (CA-316). Validation logic (required vs. invalid distinction, positive-value gating) is correctly separated per field, `perHourTotal` correctly derives instead of being stored, and `copyWith` correctly reuses the sentinel pattern so nothing regresses for the pre-existing `itemType` field. No blockers.

### Nits (non-blocking)

1. **Missing test for reset-to-null on subsequent invalid input** (`labour_cost_form_bloc.dart` line 85, mirrored in `labour_cost_form_bloc_test.dart`) — the "previously valid → now invalid" transition for `crewSize`/`hourlyRate` isn't exercised by any of the 21 tests. See inline comment for a suggested `blocTest` snippet.
2. **Asymmetric event contract** (`labour_cost_form_event.dart` line 34) — `LabourCostCrewSizeUpdated` takes a pre-parsed `double?` while `LabourCostHourlyRateUpdated` takes a raw `String`. Worth reconciling to one convention before CA-294's widget wiring starts.
3. **British/American spelling mix** (`labour_cost_form_state.dart` line 26) — `laborValue` mixes American spelling with the surrounding British "Labour" convention. Pre-existing (inherited from `LaborValue`/`LaborCalculationMethodType`), not newly introduced, but now baked into a new public event class name too. Cosmetic, optional cleanup.

Full HTML report: ~/Desktop/pr-reviews/pr-review-construculator-app-447.html

**Review by Bazabizi — APPROVED** [review]
✅ Reviewed — no blocking findings. (Reviewed in PR Review Deck.)

**Comment by ayanasamuel8** [comment]
Closing this PR — the Cost Estimation input flow it implements is being redesigned under the Estimation v2 initiative (CA-1056). The current design here doesn't match where the v2 storyboard has landed, so it's superseded rather than merged. Once the v2 design doc is finalized and broken into tickets, this scope will be re-implemented under a new ticket/PR.
